Public/Src/Cache/ContentStore/Distributed/NuCache/EventStreaming/AzureEventHubClient.cs

```diff
@@ -0,0 +1,113 @@
+        }
+
+        /// <inheritdoc />
+        public BoolResult StartProcessing(OperationContext context, EventSequencePoint sequencePoint, IPartitionReceiveHandler processor)
+        {
+            Tracer.Info(context, $"{Tracer.Name}: Initializing event processing for event hub '{_configuration.EventHubName}' and consumer group '{_configuration.ConsumerGroupName}'.");
+
+            if (_partitionReceiver == null)
+            {
+                _partitionReceiver = _eventHubClient.CreateReceiver(
+                    _configuration.ConsumerGroupName,
+                    PartitionId,
+                    GetInitialOffset(context, sequencePoint),
+                    new ReceiverOptions()
+                    {
+                        Identifier = _hostName
+                    });
+
+                _partitionReceiver.SetReceiveHandler(processor);
+            }
+
+            return BoolResult.Success;
+        }
+
+        /// <inheritdoc />
+        public BoolResult SuspendProcessing(OperationContext context)
+        {
+            // In unit tests, hangs sometimes occur for this when running multiple tests in sequence.
+            // Adding a timeout to detect when this occurs
+            _partitionReceiver?.CloseAsync().WithTimeoutAsync(TimeSpan.FromMinutes(1)).GetAwaiter().GetResult();
+            _partitionReceiver = null;
+
+            return BoolResult.Success;
+        }
+
+        /// <inheritdoc />
+        protected override async Task<BoolResult> StartupCoreAsync(OperationContext context)
+        {
+            await base.StartupCoreAsync(context).ThrowIfFailure();
+
+            var connectionStringBuilder =
+                new EventHubsConnectionStringBuilder(_configuration.EventHubConnectionString)
+                {
+                    EntityPath = _configuration.EventHubName,
+                };
+
+            // Retry behavior in the Azure Event Hubs Client Library is controlled by the RetryPolicy property on the EventHubClient class.
+            // The default policy retries with exponential backoff when Azure Event Hub returns a transient EventHubsException or an OperationCanceledException.
+            _eventHubClient = EventHubClient.CreateFromConnectionString(connectionStringBuilder.ToString());
+            _partitionSender = _eventHubClient.CreatePartitionSender(PartitionId);
+
+            return BoolResult.Success;
+        }
+
+        /// <inheritdoc />
+        protected override async Task<BoolResult> ShutdownCoreAsync(OperationContext context)
+        {
+            SuspendProcessing(context).ThrowIfFailure();
+
+            _partitionSender?.CloseAsync();
+            _eventHubClient?.CloseAsync();
+
+            return await base.ShutdownCoreAsync(context);
+        }
+
+        /// <inheritdoc />
+        public async Task SendAsync(OperationContext context, EventData eventData)
+        {
+            await _partitionSender.SendAsync(eventData);

```

Comment:
Just curious, should we keep this method async, or we can just return the result of SendAsync method.